youtube2zim/scraper.py

```diff
@@ -792,6 +792,16 @@ def make_html_files(self, actual_videos_ids):
             - videos/<videoId>/video.jpg            template
         """
 
+        def remove_unused_videos(used_videos):
+            used_video_ids = []
+            for video in used_videos:
+                used_video_ids.append(video["contentDetails"]["videoId"])
+            videos_dir_contents = [x for x in self.videos_dir.iterdir() if x.is_dir()]
+            for video_dir in videos_dir_contents:
+                if not video_dir.stem in used_video_ids:
```

Comment:
don't use `fname.stem` here as stem is the *filename without extension* part. That won't work if the videoId has a `.` in it (some does).
Instead, use `fname.name` that does what you want.